### PR TITLE
fix(current): guard get_query_var when $wp_query is null (template-switching 500)

### PR DESCRIPTION
**Pull request by superdav42**
## Summary

Fix a PHP 8+ fatal in `Current::load_currents()` that surfaces in production
as **"A network error occurred. Please check your connection and try again."**
on the customer-panel **Switch Template** page (and any other wu-ajax
endpoint that reads `is_customer_allowed()`).

## Root cause

PR #1118 added a lazy-load entry from `Current::get_customer()` /
`get_membership()` so callers that arrive before the `init`/`wp` cache
warm-up still get a populated value. That fix is correct in spirit but
exposes a latent hazard inside `load_currents()` itself: both pretty-URL
hash overrides call `get_query_var()`.

`get_query_var()` is `$wp_query->get($var, $default)`. The wu-ajax
pipeline (`Light_Ajax::process_light_ajax`) dispatches `wu_ajax_*`
handlers at `plugins_loaded` priority 20 — long before WordPress reaches
`parse_query`. At that point `$wp_query` is `null`, and PHP 8+ fatals
with `Call to a member function get() on null`. The AJAX handler dies
with HTTP 500 and no JSON body, so the JS error callback shows the
generic network-error banner.

Backtrace from a real customer (production):

```
#0 wp-includes/query.php:29  — get_query_var('site_hash')
#1 inc/class-current.php:253 — load_currents()
#2 inc/class-current.php:377 — get_customer()
#3 inc/models/class-site.php:1075 — Site->is_customer_allowed()
#4 inc/ui/class-template-switching-element.php:320 — switch_template()
...
#7 inc/class-light-ajax.php:160 — do_action('wu_ajax_wu_switch_template')
#11 wp-settings.php:593 — do_action('plugins_loaded')
```

## Fix

Compute `$query_vars_ready = did_action('parse_query') && $GLOBALS['wp_query'] instanceof WP_Query`
once at the top of `load_currents()`. When false, substitute an empty
string for both `get_query_var()` calls. `wu_request()` still picks up
`$_REQUEST` overrides if present, and the wu-ajax pipeline carries no
rewrite-driven `site_hash` / `membership_hash` anyway (it is a flat
`?wu-ajax=1` POST), so nothing functional is lost. Once WP reaches `wp`
and re-runs `load_currents()`, the guard is true and the original
behaviour is preserved verbatim.

## Test plan

- New regression test `test_load_currents_survives_when_wp_query_is_null`
  nulls `$GLOBALS['wp_query']` and asserts `load_currents()` runs to
  completion. Verified by reverting the fix and confirming the test
  reproduces the production fatal exactly:
  > `Error: Call to a member function get() on null at class-current.php:253`
- Full `Current_Test` suite: **11 tests, 16 assertions, all pass**.
- PHPCS: clean.
- PHPStan: clean.
- Other test failures in the suite are pre-existing and unrelated
  (verified by running the same scope on `origin/main`).

## Files changed

- `inc/class-current.php` — guard both `get_query_var()` calls.
- `tests/WP_Ultimo/Current_Test.php` — regression test.

Resolves the customer-reported 500 / "A network error occurred" banner
on the customer-panel template-switching page reported after #1118
landed.

---
[aidevops.sh](https://aidevops.sh) v3.14.75 plugin for [OpenCode](https://opencode.ai) v1.14.33 with claude-sonnet-4-6 spent 1d 23h on this as a headless worker.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed potential errors that could occur during specific WordPress initialization scenarios, improving system stability.

* **Performance**
  * Optimized object loading to prevent unnecessary reprocessing in administrative and background request contexts.

### CODE REVIEW

**Comment by superdav42** [comment]
## Summary
Fix a PHP 8+ fatal in `Current::load_currents()` that surfaces in production
as **"A network error occurred. Please check your connection and try again."**
on the customer-panel **Switch Template** page (and any other wu-ajax
endpoint that reads `is_customer_allowed()`).
## Root cause
PR #1118 added a lazy-load entry from `Current::get_customer()` /
`get_membership()` so callers that arrive before the `init`/`wp` cache
warm-up still get a populated value. That fix is correct in spirit but
exposes a latent hazard inside `load_currents()` itself: both pretty-URL
hash overrides call `get_query_var()`.
`get_query_var()` is `$wp_query->get($var, $default)`. The wu-ajax
pipeline (`Light_Ajax::process_light_ajax`) dispatches `wu_ajax_*`
handlers at `plugins_loaded` priority 20 — long before WordPress reaches
`parse_query`. At that point `$wp_query` is `null`, and PHP 8+ fatals
with `Call to a member function get() on null`. The AJAX handler dies
with HTTP 500 and no JSON body, so the JS error callback shows the
generic network-error banner.
Backtrace from a real customer (production):
```
#0 wp-includes/query.php:29  — get_query_var('site_hash')
#1 inc/class-current.php:253 — load_currents()
#2 inc/class-current.php:377 — get_customer()
#3 inc/models/class-site.php:1075 — Site->is_customer_allowed()
#4 inc/ui/class-template-switching-element.php:320 — switch_template()
...
#7 inc/class-light-ajax.php:160 — do_action('wu_ajax_wu_switch_template')
#11 wp-settings.php:593 — do_action('plugins_loaded')
```
## Fix
Compute `$query_vars_ready = did_action('parse_query') && $GLOBALS['wp_query'] instanceof WP_Query`
once at the top of `load_currents()`. When false, substitute an empty
string for both `get_query_var()` calls. `wu_request()` still picks up
`$_REQUEST` overrides if present, and the wu-ajax pipeline carries no
rewrite-driven `site_hash` / `membership_hash` anyway (it is a flat
`?wu-ajax=1` POST), so nothing functional is lost. Once WP reaches `wp`
and re-runs `load_currents()`, the guard is true and the original
behaviour is preserved verbatim.
## Test plan
- New regression test `test_load_currents_survives_when_wp_query_is_null`
  nulls `$GLOBALS['wp_query']` and asserts `load_currents()` runs to
  completion. Verified by reverting the fix and confirming the test
  reproduces the production fatal exactly:
  > `Error: Call to a member function get() on null at class-current.php:253`
- Full `Current_Test` suite: **11 tests, 16 assertions, all pass**.
- PHPCS: clean.
- PHPStan: clean.
- Other test failures in the suite are pre-existing and unrelated
  (verified by running the same scope on `origin/main`).
## Files changed
- `inc/class-current.php` — guard both `get_query_var()` calls.
- `tests/WP_Ultimo/Current_Test.php` — regression test.
Resolves the customer-reported 500 / "A network error occurred" banner
on the customer-panel template-switching page reported after #1118
landed.

---
[aidevops.sh](https://aidevops.sh) v3.14.75 plugin for [OpenCode](https://opencode.ai) v1.14.33 with claude-sonnet-4-6 spent 1d 23h on this as a headless worker.

---
Merged via PR #1127 to main.
_Merged by deterministic merge pass (pulse-wrapper.sh)._
